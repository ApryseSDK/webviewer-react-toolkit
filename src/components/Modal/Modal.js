'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Modal = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var hooks_1 = require('../../hooks');
var utils_1 = require('../../utils');
var ButtonGroup_1 = require('../ButtonGroup');
var FocusTrap_1 = require('../FocusTrap');
var Icon_1 = require('../Icon');
var IconButton_1 = require('../IconButton');
var Overlay_1 = require('../Overlay');
/* eslint-disable jsx-a11y/interactive-supports-focus, jsx-a11y/click-events-have-key-events */
var Modal = function(_a) {
  var closeOnBackgroundClick = _a.closeOnBackgroundClick,
    closeOnEscape = _a.closeOnEscape,
    heading = _a.heading,
    open = _a.open,
    onClose = _a.onClose,
    fullWidth = _a.fullWidth,
    noUnmount = _a.noUnmount,
    wrapperClassName = _a.wrapperClassName,
    _b = _a.closeLabel,
    closeLabel = _b === void 0 ? 'Close' : _b,
    children = _a.children,
    buttonGroup = _a.buttonGroup,
    className = _a.className,
    _c = _a.role,
    role = _c === void 0 ? 'dialog' : _c,
    _d = _a['aria-modal'],
    ariaModal = _d === void 0 ? true : _d,
    props = tslib_1.__rest(_a, [
      'closeOnBackgroundClick',
      'closeOnEscape',
      'heading',
      'open',
      'onClose',
      'fullWidth',
      'noUnmount',
      'wrapperClassName',
      'closeLabel',
      'children',
      'buttonGroup',
      'className',
      'role',
      'aria-modal',
    ]);
  var mounted = (0, hooks_1.useUnmountDelay)(open).mounted;
  var headingId = (0, react_1.useMemo)(function() {
    return (0, utils_1.getStringId)('modal_heading');
  }, []);
  var bodyId = (0, react_1.useMemo)(function() {
    return (0, utils_1.getStringId)('modal_body');
  }, []);
  (0, react_1.useEffect)(
    function() {
      if (open && closeOnEscape && onClose) {
        var listener_1 = function(event) {
          if (event.key === 'Escape') onClose(event);
        };
        window.addEventListener('keydown', listener_1);
        return function() {
          return window.removeEventListener('keydown', listener_1);
        };
      }
      return;
    },
    [closeOnEscape, onClose, open],
  );
  var backgroundIsButton = !!(open && closeOnBackgroundClick && onClose);
  var handleBackgroundClick = (0, react_1.useCallback)(
    function(event) {
      if (!backgroundIsButton) return;
      if (event.currentTarget !== event.target) return;
      onClose === null || onClose === void 0 ? void 0 : onClose(event);
    },
    [backgroundIsButton, onClose],
  );
  var modalWrapperClass = (0, classnames_1.default)(
    'ui__base ui__modal__wrapper',
    {
      'ui__modal__wrapper--closed': !open,
      'ui__modal__wrapper--fullWidth': fullWidth,
    },
    wrapperClassName,
  );
  var modalClass = (0, classnames_1.default)(
    'ui__modal',
    { 'ui__modal--hidden': noUnmount && !mounted },
    className,
  );
  var bodyClass = (0, classnames_1.default)('ui__modal__body', {
    'ui__modal__body--noButton': !buttonGroup,
  });
  return react_1.default.createElement(
    Overlay_1.Overlay,
    null,
    react_1.default.createElement(
      'div',
      {
        role: backgroundIsButton ? 'button' : undefined,
        className: modalWrapperClass,
        onClick: handleBackgroundClick,
      },
      noUnmount || mounted
        ? react_1.default.createElement(
            'div',
            { className: 'ui__modal__paddingFix' },
            react_1.default.createElement(
              FocusTrap_1.FocusTrap,
              { focusLastOnUnlock: true, locked: open },
              react_1.default.createElement(
                'div',
                tslib_1.__assign(
                  {
                    'aria-labelledby': heading ? headingId : undefined,
                    'aria-describedby': bodyId,
                  },
                  props,
                  {
                    className: modalClass,
                    role: role,
                    'aria-modal': ariaModal,
                  },
                ),
                react_1.default.createElement(
                  'div',
                  { className: 'ui__modal__top' },
                  react_1.default.createElement(
                    'div',
                    { className: 'ui__modal__top__heading', id: headingId },
                    heading,
                  ),
                  onClose
                    ? react_1.default.createElement(
                        IconButton_1.IconButton,
                        {
                          className: 'ui__modal__top__close',
                          onClick: onClose,
                          'aria-label': closeLabel,
                        },
                        react_1.default.createElement(Icon_1.Icon, {
                          icon: 'Close',
                        }),
                      )
                    : undefined,
                ),
                react_1.default.createElement(
                  'div',
                  { className: bodyClass, id: bodyId },
                  children,
                ),
                buttonGroup
                  ? react_1.default.createElement(
                      ButtonGroup_1.ButtonGroup,
                      { className: 'ui__modal__buttonGroup' },
                      buttonGroup,
                    )
                  : undefined,
              ),
            ),
          )
        : undefined,
    ),
  );
};
exports.Modal = Modal;
//# sourceMappingURL=Modal.js.map
