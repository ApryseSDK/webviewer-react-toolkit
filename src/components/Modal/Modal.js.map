{"version":3,"file":"Modal.js","sourceRoot":"","sources":["Modal.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAUe;AACf,qCAA8C;AAC9C,qCAA0C;AAC1C,8CAA6C;AAC7C,0CAAyC;AACzC,gCAA+B;AAC/B,4CAA2C;AAC3C,sCAAqC;AA4DrC,+FAA+F;AAExF,IAAM,KAAK,GAAmB,UAAC,EAgBrC;IAfC,IAAA,sBAAsB,4BAAA,EACtB,aAAa,mBAAA,EACb,OAAO,aAAA,EACP,IAAI,UAAA,EACJ,OAAO,aAAA,EACP,SAAS,eAAA,EACT,SAAS,eAAA,EACT,gBAAgB,sBAAA,EAChB,kBAAoB,EAApB,UAAU,mBAAG,OAAO,KAAA,EACpB,QAAQ,cAAA,EACR,WAAW,iBAAA,EACX,SAAS,eAAA,EACT,YAAe,EAAf,IAAI,mBAAG,QAAQ,KAAA,EACf,qBAA8B,EAAhB,SAAS,mBAAG,IAAI,KAAA,EAC3B,KAAK,sBAf4B,mMAgBrC,CADS;IAEA,IAAA,OAAO,GAAK,IAAA,uBAAe,EAAC,IAAI,CAAC,QAA1B,CAA2B;IAE1C,IAAM,SAAS,GAAG,IAAA,eAAO,EAAC,cAAM,OAAA,IAAA,mBAAW,EAAC,eAAe,CAAC,EAA5B,CAA4B,EAAE,EAAE,CAAC,CAAC;IAClE,IAAM,MAAM,GAAG,IAAA,eAAO,EAAC,cAAM,OAAA,IAAA,mBAAW,EAAC,YAAY,CAAC,EAAzB,CAAyB,EAAE,EAAE,CAAC,CAAC;IAE5D,IAAA,iBAAS,EAAC;QACR,IAAI,IAAI,IAAI,aAAa,IAAI,OAAO,EAAE;YACpC,IAAM,UAAQ,GAAG,UAAC,KAAoB;gBACpC,IAAI,KAAK,CAAC,GAAG,KAAK,QAAQ;oBAAE,OAAO,CAAC,KAAK,CAAC,CAAC;YAC7C,CAAC,CAAC;YACF,MAAM,CAAC,gBAAgB,CAAC,SAAS,EAAE,UAAQ,CAAC,CAAC;YAC7C,OAAO,cAAM,OAAA,MAAM,CAAC,mBAAmB,CAAC,SAAS,EAAE,UAAQ,CAAC,EAA/C,CAA+C,CAAC;SAC9D;QACD,OAAO;IACT,CAAC,EAAE,CAAC,aAAa,EAAE,OAAO,EAAE,IAAI,CAAC,CAAC,CAAC;IAEnC,IAAM,kBAAkB,GAAG,CAAC,CAAC,CAAC,IAAI,IAAI,sBAAsB,IAAI,OAAO,CAAC,CAAC;IAEzE,IAAM,qBAAqB,GAAG,IAAA,mBAAW,EACvC,UAAC,KAAK;QACJ,IAAI,CAAC,kBAAkB;YAAE,OAAO;QAChC,IAAI,KAAK,CAAC,aAAa,KAAK,KAAK,CAAC,MAAM;YAAE,OAAO;QACjD,OAAO,aAAP,OAAO,uBAAP,OAAO,CAAG,KAAK,CAAC,CAAC;IACnB,CAAC,EACD,CAAC,kBAAkB,EAAE,OAAO,CAAC,CAC9B,CAAC;IAEF,IAAM,iBAAiB,GAAG,IAAA,oBAAU,EAClC,6BAA6B,EAC7B;QACE,4BAA4B,EAAE,CAAC,IAAI;QACnC,+BAA+B,EAAE,SAAS;KAC3C,EACD,gBAAgB,CACjB,CAAC;IAEF,IAAM,UAAU,GAAG,IAAA,oBAAU,EAAC,WAAW,EAAE,EAAE,mBAAmB,EAAE,SAAS,IAAI,CAAC,OAAO,EAAE,EAAE,SAAS,CAAC,CAAC;IAEtG,IAAM,SAAS,GAAG,IAAA,oBAAU,EAAC,iBAAiB,EAAE;QAC9C,2BAA2B,EAAE,CAAC,WAAW;KAC1C,CAAC,CAAC;IAEH,OAAO,CACL,8BAAC,iBAAO;QACN,uCACE,IAAI,EAAE,kBAAkB,CAAC,CAAC,CAAC,QAAQ,CAAC,CAAC,CAAC,SAAS,EAC/C,SAAS,EAAE,iBAAiB,EAC5B,OAAO,EAAE,qBAAqB,IAE7B,SAAS,IAAI,OAAO,CAAC,CAAC,CAAC,CACtB,uCAAK,SAAS,EAAC,uBAAuB;YACpC,8BAAC,qBAAS,IAAC,iBAAiB,QAAC,MAAM,EAAE,IAAI;gBACvC,2EACmB,OAAO,CAAC,CAAC,CAAC,SAAS,CAAC,CAAC,CAAC,SAAS,sBAC9B,MAAM,IACpB,KAAK,IACT,SAAS,EAAE,UAAU,EACrB,IAAI,EAAE,IAAI,gBACE,SAAS;oBAErB,uCAAK,SAAS,EAAC,gBAAgB;wBAC7B,uCAAK,SAAS,EAAC,yBAAyB,EAAC,EAAE,EAAE,SAAS,IACnD,OAAO,CACJ;wBACL,OAAO,CAAC,CAAC,CAAC,CACT,8BAAC,uBAAU,IAAC,SAAS,EAAC,uBAAuB,EAAC,OAAO,EAAE,OAAO,gBAAc,UAAU;4BACpF,8BAAC,WAAI,IAAC,IAAI,EAAC,OAAO,GAAG,CACV,CACd,CAAC,CAAC,CAAC,SAAS,CACT;oBACN,uCAAK,SAAS,EAAE,SAAS,EAAE,EAAE,EAAE,MAAM,IAClC,QAAQ,CACL;oBACL,WAAW,CAAC,CAAC,CAAC,8BAAC,yBAAW,IAAC,SAAS,EAAC,wBAAwB,IAAE,WAAW,CAAe,CAAC,CAAC,CAAC,SAAS,CAClG,CACI,CACR,CACP,CAAC,CAAC,CAAC,SAAS,CACT,CACE,CACX,CAAC;AACJ,CAAC,CAAC;AAlGW,QAAA,KAAK,SAkGhB"}