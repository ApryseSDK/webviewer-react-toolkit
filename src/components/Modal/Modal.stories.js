'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.UnmountExample = exports.Scrolling = exports.WithButtonGroup = exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importStar(require('react'));
var action_1 = require('../../storybook-helpers/action/action');
var Button_1 = require('../Button');
var Modal_1 = require('../Modal');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/Modal',
  component: Modal_1.Modal,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(
    Modal_1.Modal,
    {
      open: (0, addon_knobs_1.boolean)('open', true),
      closeOnBackgroundClick: (0, addon_knobs_1.boolean)(
        'closeOnBackgroundClick',
        false,
      ),
      closeOnEscape: (0, addon_knobs_1.boolean)('closeOnEscape', false),
      heading: (0, addon_knobs_1.text)('heading', 'Modal heading'),
      fullWidth: (0, addon_knobs_1.boolean)('fullWidth', false),
      onClose: (0, addon_knobs_1.boolean)('has onClose', true)
        ? (0, action_1.action)('onClose')
        : undefined,
    },
    (0, addon_knobs_1.text)(
      'children',
      'Modal children contains the description of what the modal is shown for. ' +
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod ' +
        'tempor incididunt ut labore et dolore magna aliqua.',
    ) || 'There must be some child!',
  );
};
exports.Basic = Basic;
var WithButtonGroup = function() {
  return react_1.default.createElement(
    Modal_1.Modal,
    {
      open: (0, addon_knobs_1.boolean)('open', true),
      closeOnBackgroundClick: (0, addon_knobs_1.boolean)(
        'closeOnBackgroundClick',
        false,
      ),
      closeOnEscape: (0, addon_knobs_1.boolean)('closeOnEscape', false),
      heading: (0, addon_knobs_1.text)('heading', 'Modal heading'),
      fullWidth: (0, addon_knobs_1.boolean)('fullWidth', false),
      onClose: (0, addon_knobs_1.boolean)('has onClose', true)
        ? (0, action_1.action)('onClose')
        : undefined,
      buttonGroup: react_1.default.createElement(
        react_1.default.Fragment,
        null,
        react_1.default.createElement(Button_1.Button, null, 'Accept'),
        react_1.default.createElement(
          Button_1.Button,
          { buttonStyle: 'outline' },
          'Cancel',
        ),
      ),
    },
    (0, addon_knobs_1.text)(
      'children',
      'Modal children contains the description of what the modal is shown for. ' +
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod ' +
        'tempor incididunt ut labore et dolore magna aliqua.',
    ) || 'There must be some child!',
  );
};
exports.WithButtonGroup = WithButtonGroup;
var lorem =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Cursus eget nunc scelerisque viverra. In dictum non consectetur a erat nam at. Euismod quis viverra nibh cras pulvinar mattis. Quis auctor elit sed vulputate mi. Turpis egestas sed tempus urna et pharetra pharetra massa massa. Magna fermentum iaculis eu non diam. Pellentesque sit amet porttitor eget dolor morbi non arcu risus. Blandit libero volutpat sed cras ornare arcu. Diam phasellus vestibulum lorem sed risus ultricies. Egestas purus viverra accumsan in nisl nisi.\n\nDictum fusce ut placerat orci nulla pellentesque dignissim enim. Lectus nulla at volutpat diam ut. Tellus elementum sagittis vitae et leo duis ut diam. Pellentesque adipiscing commodo elit at imperdiet dui accumsan sit. Enim eu turpis egestas pretium aenean pharetra. Sed viverra tellus in hac. Velit sed ullamcorper morbi tincidunt ornare. Consequat mauris nunc congue nisi vitae suscipit. Blandit massa enim nec dui nunc mattis enim ut. Sed augue lacus viverra vitae congue eu consequat ac. Iaculis at erat pellentesque adipiscing. Amet risus nullam eget felis eget nunc. Sodales ut etiam sit amet nisl purus in mollis. Proin sed libero enim sed faucibus turpis. Sapien nec sagittis aliquam malesuada bibendum arcu. Bibendum at varius vel pharetra vel turpis nunc. Pellentesque habitant morbi tristique senectus. Euismod elementum nisi quis eleifend. Augue ut lectus arcu bibendum at varius vel pharetra.\n\nAliquam nulla facilisi cras fermentum odio. Est velit egestas dui id ornare. Nibh sit amet commodo nulla facilisi nullam vehicula ipsum. Sit amet risus nullam eget felis eget nunc lobortis. Neque ornare aenean euismod elementum nisi quis eleifend. Sed ullamcorper morbi tincidunt ornare. Non enim praesent elementum facilisis leo vel fringilla. Amet dictum sit amet justo donec enim diam. Consectetur adipiscing elit duis tristique sollicitudin. Consequat id porta nibh venenatis cras sed felis. Viverra tellus in hac habitasse. Aliquam ut porttitor leo a diam sollicitudin tempor id eu. Quis ipsum suspendisse ultrices gravida dictum fusce ut placerat. Faucibus purus in massa tempor nec. Interdum posuere lorem ipsum dolor sit amet consectetur. Facilisi cras fermentum odio eu feugiat. Arcu ac tortor dignissim convallis aenean et tortor at. Nascetur ridiculus mus mauris vitae ultricies leo.\n\nAdipiscing elit duis tristique sollicitudin nibh sit. Vel elit scelerisque mauris pellentesque pulvinar pellentesque habitant morbi. Adipiscing elit duis tristique sollicitudin nibh. Aliquam etiam erat velit scelerisque in dictum. Tristique magna sit amet purus gravida quis blandit turpis cursus. Fermentum leo vel orci porta non. Et sollicitudin ac orci phasellus. Arcu dui vivamus arcu felis. Dignissim suspendisse in est ante in nibh mauris cursus. Sit amet nisl purus in mollis nunc sed id. Viverra nibh cras pulvinar mattis nunc sed blandit libero. Duis convallis convallis tellus id. Mauris pharetra et ultrices neque ornare aenean euismod. Aliquet porttitor lacus luctus accumsan tortor posuere.\n\nTortor pretium viverra suspendisse potenti nullam ac tortor vitae. Eu turpis egestas pretium aenean. Justo eget magna fermentum iaculis eu non diam. Sagittis id consectetur purus ut faucibus pulvinar elementum integer. Malesuada fames ac turpis egestas sed tempus urna et. Cras sed felis eget velit aliquet sagittis id consectetur. Duis tristique sollicitudin nibh sit amet commodo nulla facilisi nullam. Malesuada fames ac turpis egestas maecenas pharetra. Magnis dis parturient montes nascetur ridiculus mus mauris. Molestie ac feugiat sed lectus vestibulum mattis. Ac orci phasellus egestas tellus. Lectus vestibulum mattis ullamcorper velit sed ullamcorper morbi tincidunt. Morbi blandit cursus risus at ultrices mi tempus imperdiet nulla. Morbi non arcu risus quis varius. Pulvinar proin gravida hendrerit lectus a. Id velit ut tortor pretium viverra. Mauris pellentesque pulvinar pellentesque habitant morbi. Etiam non quam lacus suspendisse faucibus interdum. Aliquam ultrices sagittis orci a scelerisque purus semper eget. Accumsan lacus vel facilisis volutpat.\n\nTellus elementum sagittis vitae et leo duis ut diam. Ut eu sem integer vitae justo eget magna. Eu volutpat odio facilisis mauris sit amet massa. Enim nec dui nunc mattis enim. Varius duis at consectetur lorem donec massa sapien. Molestie nunc non blandit massa enim nec. Id donec ultrices tincidunt arcu non sodales neque sodales. Lorem ipsum dolor sit amet consectetur adipiscing elit duis. Sit amet nisl suscipit adipiscing. Mauris commodo quis imperdiet massa tincidunt nunc pulvinar sapien. Facilisis magna etiam tempor orci eu lobortis elementum. Cras sed felis eget velit aliquet sagittis id consectetur.\n\nPorttitor leo a diam sollicitudin tempor id eu. Enim nec dui nunc mattis enim ut tellus elementum. Integer feugiat scelerisque varius morbi enim nunc faucibus a. Quis varius quam quisque id diam. Lectus vestibulum mattis ullamcorper velit sed ullamcorper morbi tincidunt. Auctor urna nunc id cursus. Odio ut sem nulla pharetra diam sit amet. Interdum posuere lorem ipsum dolor. Dignissim cras tincidunt lobortis feugiat vivamus at augue eget arcu. Orci nulla pellentesque dignissim enim sit amet venenatis urna cursus. Libero id faucibus nisl tincidunt eget nullam non. Ut tortor pretium viverra suspendisse potenti. Ullamcorper dignissim cras tincidunt lobortis feugiat. Tristique risus nec feugiat in fermentum. Donec pretium vulputate sapien nec. Est pellentesque elit ullamcorper dignissim cras.\n\nNon odio euismod lacinia at quis risus sed. In iaculis nunc sed augue lacus. Mi sit amet mauris commodo quis imperdiet. Morbi quis commodo odio aenean sed adipiscing diam donec. Pretium fusce id velit ut tortor. Enim blandit volutpat maecenas volutpat. Ultrices vitae auctor eu augue ut. Sociis natoque penatibus et magnis dis. Leo urna molestie at elementum eu. Dictum non consectetur a erat nam at lectus urna. Cursus turpis massa tincidunt dui ut ornare lectus sit amet. Proin sagittis nisl rhoncus mattis rhoncus urna neque viverra justo. Sem et tortor consequat id porta. Commodo elit at imperdiet dui accumsan sit amet nulla. Ultrices in iaculis nunc sed augue lacus. Duis convallis convallis tellus id interdum velit laoreet. Duis at consectetur lorem donec massa sapien. Sit amet justo donec enim. Orci eu lobortis elementum nibh.\n\nNisl nunc mi ipsum faucibus. Arcu dui vivamus arcu felis bibendum. Rhoncus urna neque viverra justo nec ultrices dui sapien. Proin fermentum leo vel orci porta non pulvinar neque. Tincidunt eget nullam non nisi est sit. Tempor nec feugiat nisl pretium fusce id velit ut tortor. Faucibus purus in massa tempor nec feugiat nisl pretium. Amet facilisis magna etiam tempor orci eu. Morbi tempus iaculis urna id volutpat lacus laoreet. Sem integer vitae justo eget magna fermentum. Semper risus in hendrerit gravida. Tellus pellentesque eu tincidunt tortor aliquam nulla facilisi cras fermentum. Tellus in hac habitasse platea. Vitae sapien pellentesque habitant morbi tristique senectus et netus et.\n\nMaecenas volutpat blandit aliquam etiam erat velit. Arcu felis bibendum ut tristique et egestas. Varius vel pharetra vel turpis nunc eget lorem. In nulla posuere sollicitudin aliquam ultrices sagittis orci a scelerisque. Ultrices dui sapien eget mi proin. Volutpat ac tincidunt vitae semper quis lectus nulla at volutpat. Sociis natoque penatibus et magnis dis. Vel eros donec ac odio. Pharetra sit amet aliquam id. Ipsum nunc aliquet bibendum enim facilisis gravida neque convallis a. Dictum sit amet justo donec enim. Proin fermentum leo vel orci porta. Et magnis dis parturient montes nascetur ridiculus.\n';
var Scrolling = function() {
  return react_1.default.createElement(
    Modal_1.Modal,
    {
      open: (0, addon_knobs_1.boolean)('open', true),
      closeOnBackgroundClick: (0, addon_knobs_1.boolean)(
        'closeOnBackgroundClick',
        false,
      ),
      closeOnEscape: (0, addon_knobs_1.boolean)('closeOnEscape', false),
      heading: (0, addon_knobs_1.text)('heading', 'Modal heading'),
      fullWidth: (0, addon_knobs_1.boolean)('fullWidth', false),
      onClose: (0, addon_knobs_1.boolean)('has onClose', true)
        ? (0, action_1.action)('onClose')
        : undefined,
      buttonGroup: react_1.default.createElement(
        react_1.default.Fragment,
        null,
        react_1.default.createElement(Button_1.Button, null, 'Accept'),
        react_1.default.createElement(
          Button_1.Button,
          { buttonStyle: 'outline' },
          'Cancel',
        ),
      ),
    },
    react_1.default.createElement(
      'p',
      { style: { whiteSpace: 'pre-wrap' } },
      lorem,
    ),
  );
};
exports.Scrolling = Scrolling;
var Timer = function() {
  var _a = (0, react_1.useState)(0),
    seconds = _a[0],
    setSeconds = _a[1];
  (0, react_1.useEffect)(function() {
    var increment = function() {
      return setSeconds(function(prev) {
        return prev + 1;
      });
    };
    var timer = window.setInterval(increment, 1000);
    return function() {
      window.clearInterval(timer);
    };
  }, []);
  return react_1.default.createElement(
    'p',
    null,
    'Time passed: ',
    seconds,
    ' second',
    seconds === 1 ? '' : 's',
  );
};
var UnmountExample = function() {
  var noUnmount = (0, addon_knobs_1.boolean)('noUnmount', false);
  return react_1.default.createElement(
    Modal_1.Modal,
    {
      open: (0, addon_knobs_1.boolean)('open', true),
      onClose: (0, action_1.action)('onClose'),
      noUnmount: noUnmount,
      heading: noUnmount
        ? 'noUnmount true, time will persist'
        : 'noUnmount false, time will reset',
    },
    react_1.default.createElement(Timer, null),
  );
};
exports.UnmountExample = UnmountExample;
//# sourceMappingURL=Modal.stories.js.map
