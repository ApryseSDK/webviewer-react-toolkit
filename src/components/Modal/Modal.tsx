import classnames from 'classnames';
import React, { FC, HTMLAttributes, MouseEvent, ReactNode, useEffect, useMemo } from 'react';
import { useUnmountDelay } from '../../hooks';
import { Close } from '../../icons';
import { ButtonGroup } from '../ButtonGroup';
import { FocusTrap } from '../FocusTrap';
import { IconButton } from '../IconButton';
import { Overlay } from '../Overlay';
import { getStringId } from '../../utils';

export interface ModalProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * Heading to display at the top of the modal.
   */
  heading: ReactNode;
  /**
   * The primary body content of the modal.
   */
  children: ReactNode;
  /**
   * Content, generally buttons, can be passed to this prop in order to render
   * them within a special button group section at the bottom of the modal.
   */
  buttonGroup?: ReactNode;
  /**
   * If given, modal will fire `onClose` (if provided) when the background is
   * clicked.
   */
  closeOnBackgroundClick?: boolean;
  /**
   * If given, modal will fire `onClose` (if provided) when escape key is
   * pressed.
   */
  closeOnEscape?: boolean;
  /**
   * Modal opens when true.
   */
  open?: boolean;
  /**
   * If given, modal will have a close button, and clicking it will fire this
   * callback function.
   * @param event Either a mouse event (background or close button clicked) or
   * a keyboard event (escape key pressed).
   */
  onClose?(event: KeyboardEvent | MouseEvent): void;
}

/* eslint-disable jsx-a11y/interactive-supports-focus, jsx-a11y/click-events-have-key-events */

export const Modal: FC<ModalProps> = ({
  closeOnBackgroundClick,
  closeOnEscape,
  heading,
  open,
  onClose,
  children,
  buttonGroup,
  className,

  ...props
}) => {
  const { mounted } = useUnmountDelay(open);

  const headingId = useMemo(() => getStringId('modal_heading', 8), []);
  const bodyId = useMemo(() => getStringId('modal_body', 8), []);

  useEffect(() => {
    if (open && closeOnEscape && onClose) {
      const listener = (event: KeyboardEvent) => {
        if (event.key === 'Escape') onClose(event);
      };
      window.addEventListener('keydown', listener);
      return () => window.removeEventListener('keydown', listener);
    }
    return;
  }, [closeOnEscape, onClose, open]);

  const backgroundIsButton = !!(closeOnBackgroundClick && onClose);

  const modalWrapperClass = classnames('ui__base ui__modal__wrapper', {
    'ui__modal__wrapper--closed': !open,
  });

  const modalClass = classnames('ui__modal', className);

  const bodyClass = classnames('ui__modal__body', {
    'ui__modal__body--noButton': !buttonGroup,
  });

  return (
    <Overlay>
      <div
        role={backgroundIsButton ? 'button' : undefined}
        className={modalWrapperClass}
        onClick={backgroundIsButton ? onClose : undefined}
      >
        {mounted ? (
          <FocusTrap focusLastOnUnlock locked>
            <div
              {...props}
              className={modalClass}
              role="dialog"
              aria-modal={true}
              aria-labelledby={headingId}
              aria-describedby={bodyId}
            >
              <div className="ui__modal__top">
                <div className="ui__modal__top__heading" id={headingId}>
                  {heading}
                </div>
                {onClose ? (
                  <IconButton className="ui__modal__top__close" onClick={onClose} aria-label="Close">
                    <Close />
                  </IconButton>
                ) : (
                  undefined
                )}
              </div>
              <div className={bodyClass} id={bodyId}>
                {children}
              </div>
              {buttonGroup ? <ButtonGroup className="ui__modal__buttonGroup">{buttonGroup}</ButtonGroup> : undefined}
            </div>
          </FocusTrap>
        ) : (
          undefined
        )}
      </div>
    </Overlay>
  );
};
