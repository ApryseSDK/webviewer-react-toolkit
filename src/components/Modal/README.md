Modals provide the ability to add confirmation or display important information
to the user. These can be easily dismissed by adding an `onClose`, as well as
`closeOnBackgroundClick` and `closeOnEscape` at your discretion.

Alternatively, they can enforce that the user completes the required action by
removing the `onClose` and adding your own confirmation buttons to the
`buttonGroup`.
