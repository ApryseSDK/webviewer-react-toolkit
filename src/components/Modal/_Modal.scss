.ui__modal__wrapper {
  padding: $padding;
  @include absolute-fill;
  @include flex-center;
  @include opacity-transition;
  opacity: 1;
  background-color: $color-overlay-canvas;

  &--closed {
    opacity: 0;
    pointer-events: none !important;
  }
}

.ui__modal {
  background-color: $color-gray-2;
  border-radius: $border-radius;
  overflow: hidden;

  @include for-phone-only {
    width: 100%;
  }

  @include for-tablet-up {
    max-width: 500px;
  }

  &__top {
    display: flex;
    align-items: flex-start;

    &__heading {
      font-size: $font-size-large;
      flex: 1 1 auto;
      padding: $padding-medium $padding;
    }

    &__close.ui__button {
      flex-grow: 0;
      flex-shrink: 0;
      margin: calc(-1 * #{$padding-tiny});
    }
  }

  &__body {
    padding: $padding-medium $padding;
    background-color: $color-gray-1;
    border-top: 1px solid $color-gray-3;
    border-bottom: 1px solid $color-gray-3;

    &--noButton {
      border-bottom: none;
    }
  }

  &__buttonGroup {
    padding: $padding-medium $padding;
  }
}
