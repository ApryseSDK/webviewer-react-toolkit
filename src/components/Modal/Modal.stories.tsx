import { action } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import React from 'react';
import { Button } from '../Button';
import { Modal } from '../Modal';
import readme from './README.md';

export default { title: 'Components/Modal', component: Modal, parameters: { readme } };

export const Basic = () => (
  <Modal
    open={boolean('open', true)}
    closeOnBackgroundClick={boolean('closeOnBackgroundClick', false)}
    closeOnEscape={boolean('closeOnEscape', false)}
    heading={text('heading', 'Modal heading') || 'There must be some header!'}
    onClose={boolean('has onClose', true) ? action('onClose') : undefined}
  >
    {text(
      'children',
      'Modal children contains the description of what the modal is shown for. ' +
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod ' +
        'tempor incididunt ut labore et dolore magna aliqua.',
    ) || 'There must be some child!'}
  </Modal>
);

export const WithButtonGroup = () => (
  <Modal
    open={boolean('open', true)}
    closeOnBackgroundClick={boolean('closeOnBackgroundClick', false)}
    closeOnEscape={boolean('closeOnEscape', false)}
    heading={text('heading', 'Modal heading') || 'There must be some header!'}
    onClose={boolean('has onClose', true) ? action('onClose') : undefined}
    buttonGroup={
      <>
        <Button key="1">Accept</Button>
        <Button key="2" buttonStyle="outline">
          Cancel
        </Button>
      </>
    }
  >
    {text(
      'children',
      'Modal children contains the description of what the modal is shown for. ' +
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod ' +
        'tempor incididunt ut labore et dolore magna aliqua.',
    ) || 'There must be some child!'}
  </Modal>
);
