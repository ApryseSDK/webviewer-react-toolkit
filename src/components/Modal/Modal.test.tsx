import { shallow } from 'enzyme';
import React from 'react';
import { Modal } from '../Modal';

describe('Modal component', () => {
  it('renders its contents when open', () => {
    const modal = shallow(
      <Modal heading="" open>
        children
      </Modal>,
    );
    expect(modal.find('.ui__modal')).toHaveLength(1);
  });

  it('hides its contents when closed', () => {
    const modal = shallow(<Modal heading="">children</Modal>);
    expect(modal.find('.ui__modal')).toHaveLength(0);
  });

  it('snapshot renders default modal', () => {
    const modal = shallow(<Modal heading="">children</Modal>);
    expect(modal).toMatchSnapshot();
  });
});
