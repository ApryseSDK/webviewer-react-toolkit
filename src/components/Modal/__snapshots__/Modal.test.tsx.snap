// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component snapshot renders default modal 1`] = `
<Component>
  <div
    className="ui__base ui__modal__wrapper ui__modal__wrapper--closed"
  />
</Component>
`;
